t/t2712-python-cli-alloc.t

```diff
@@ -94,19 +94,12 @@ test_expect_success 'flux alloc --bg fails if rc1 fails' '
 '
 
 
-#  Running a process in the background under test_expect_success()
-#   causes a copy of the shell to be run in between flux, so the
-#   signal can't be delivered to the right PID. Running from a function
-#   seems to fix that.
-run_mini_bg() {
-	flux alloc --bg -n1 -v >sigint.log 2>&1 &
-	echo $! >sigint.pid
-}
 waitfile=$SHARNESS_TEST_SRCDIR/scripts/waitfile.lua
 test_expect_success NO_CHAIN_LINT 'flux alloc --bg can be interrupted' '
 	flux queue stop &&
-	test_when_finished "flux queue start" &&
-	run_mini_bg &&
+	test_when_finished "flux queue start"
+	flux alloc --bg -n1 -v >sigint.log 2>&1 &
+	echo $! >sigint.pid &&
```

Comment:
I'm wondering the reasoning behind changing a working test (that also had a nice comment)?
The commit message says "remove a workaround" but removing the `&&` is also just a different workaround...

Reply:
Fair point.  My assumption is that the test did not work b/c the author did not know about the && chain race.  But it is possible there's another subtlety in there I'm unaware of.  I'll put back the original.